Backend/Remora.Discord.Rest/API/Channels/DiscordRestChannelAPI.cs

```diff
@@ -355,7 +357,7 @@ public virtual Task<Result> DeleteChannelAsync
     )
     {
         var hasAny = around.HasValue || before.HasValue || after.HasValue;
-        var hasStrictlyOne = around.HasValue ^ before.HasValue ^ after.HasValue;
+        var hasStrictlyOne = (around.HasValue ^ before.HasValue ^ after.HasValue) && (!around.HasValue || !before.HasValue || !after.HasValue);
```

Comment:
This check seems redundant, unless I'm missing something. Can you elaborate?

Reply:
The second part is required to also disallow having all 3 being set as
`a ^ b ^ c` for `a = b = c = 1/true` would result in `1/true`.

Reply:
Ah, got you. Nice catch.